src/js/node/_http_incoming.ts

```diff
@@ -341,95 +383,369 @@ const IncomingMessagePrototype = {
+      return;
+    }
+    this.__socket = value;
+    return;
   },
-  get httpVersionMinor() {
-    const version = this.httpVersion;
-    if (version.endsWith(".1")) {
-      return 1;
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "rawTrailers", {
+  get() {
+    if (this[kBunServer]) {
+      return [];
     }
-    return 0;
+    return this.__rawTrailers;
   },
-  set httpVersionMinor(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      return;
+    }
+    this.__rawTrailers = value;
+    return;
   },
-  get rawTrailers() {
-    return [];
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "aborted", {
+  get() {
+    if (this[kBunServer]) {
+      return this[abortedSymbol];
+    }
+    return this.__aborted;
   },
-  set rawTrailers(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      this[abortedSymbol] = value;
+      return;
+    }
+    this.__aborted = value;
+    return;
   },
-  get trailers() {
-    return kEmptyObject;
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "statusCode", {
+  get() {
+    if (this[kBunServer]) {
+      return this[statusCodeSymbol];
+    }
+    return this.__statusCode;
   },
-  set trailers(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      if (!(value in STATUS_CODES)) return;
+      this[statusCodeSymbol] = value;
+      return;
+    }
+    this.__statusCode = value;
+    return;
   },
-  setTimeout(msecs, callback) {
-    void this.take;
-    const req = this[kHandle] || this[webRequestOrResponse];
+});
 
-    if (req) {
-      setRequestTimeout(req, Math.ceil(msecs / 1000));
-      if (typeof callback === "function") this.once("timeout", callback);
+ObjectDefineProperty(IncomingMessage.prototype, "statusMessage", {
+  get() {
+    if (this[kBunServer]) {
+      return this[statusMessageSymbol];
     }
-    return this;
+    return this.__statusMessage;
   },
-  get socket() {
-    return (this[fakeSocketSymbol] ??= new FakeSocket(this));
-  },
-  set socket(value) {
-    this[fakeSocketSymbol] = value;
+  set(value) {
+    if (this[kBunServer]) {
+      this[statusMessageSymbol] = value;
+      return;
+    }
+    this.__statusMessage = value;
+    return;
   },
-} satisfies typeof import("node:http").IncomingMessage.prototype;
-IncomingMessage.prototype = IncomingMessagePrototype;
-$setPrototypeDirect.$call(IncomingMessage, Readable);
-
-function requestHasNoBody(method, req) {
-  if ("GET" === method || "HEAD" === method || "TRACE" === method || "CONNECT" === method || "OPTIONS" === method)
-    return true;
-  const headers = req?.headers;
-  const contentLength = headers?.["content-length"];
-  if (!parseInt(contentLength, 10)) return true;
-
-  return false;
+});
+
+IncomingMessage.prototype._construct = function (callback) {
+  if (this[kBunServer]) {
+    // TODO: streaming
+    const type = this[typeSymbol];
+    if (type === NodeHTTPIncomingRequestType.FetchResponse) {
```

Comment:
Do we still need to support `NodeHTTPIncomingRequestType.FetchResponse` since it no longer uses `Response` in either the server or the client code paths? If so, should this code path be deleted?